infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -356,6 +366,30 @@ public static void configureBootstrapOptions(@Nonnull ServerContext context,
             .childOption(ChannelOption.ALLOCATOR, PooledByteBufAllocator.DEFAULT);
     }
 
+    /** Get the type of {@link ServerChannel} to use.
+     *
+     * @param context       The {@link ServerContext} to use.
+     * @return              The server channel type.
+     */
+    public static Class<? extends ServerSocketChannel> getServerChannelType(
+            @Nonnull ServerContext context) {
+        final String type = context.getServerConfig(String.class, "--implementation");
+        switch (type) {
+            case "auto":
+                return Epoll.isAvailable() ? EpollServerSocketChannel.class :
+                    KQueue.isAvailable() ? KQueueServerSocketChannel.class :
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)